provider/awsv4.go

```diff
@@ -0,0 +1,39 @@
+package provider
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+	"net/http"
+)
+
+var emptyStringSHA256 = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
```

Comment:
Thanks, can we have the logic so that `emptyStringSHA256` is generated dynamically ?

Reply:
Thanks for your contribution @vasyaxparfenov, just one comment rest the PR LGTM.

Reply:
Done